server/src/test/java/org/opensearch/common/time/DateFormattersTests.java

```diff
@@ -461,6 +461,48 @@ public void testIso8601Parsing() {
         formatter.format(formatter.parse("2018-05-15T17:14:56,123456789+01:00"));
     }
 
+    public void testRFC339Parsing() {
```

Comment:
Instead of auto generating test cases which itself will be a blackbox for anyone understanding the functionality of parser, I've added appropriate test cases with all the date formats applicable to rfc3339 which is a finite subset here. I've added the test to compare the results with existing opensearch parsers which themselves are based on jdk parsers